test/controllers/api/v2/upstream_subscriptions_controller_test.rb

```diff
@@ -38,5 +40,30 @@ def test_index_protected
         get :index, params: { :organization_id => @organization.id }
       end
     end
+
+    def test_create
+      UpstreamConsumer.stubs(:bind_entitlements).returns([])
+      pool = {id: '3', quantity: 6}
+      params = { organization_id: @organization.id, pools: [pool] }
+
+      assert_async_task ::Actions::Katello::UpstreamSubscriptions::BindEntitlements do |organization_id, pools|
+        assert_equal(organization_id, @organization.id.to_s)
+        assert_equal(pools, [ActionController::Parameters.new(pool).permit!])
```

Comment:
Ah! We had some issues in the rails 5.1 upgrade with sending ActionController::Parameters outside of controllers because they do not have the same API as a hash.

I recommend that we have a convention of calling .to_h on hash parameters being sent outside of the controller into any other object just to Be Safe.

Then you could assert_equal(pools, your_expected_pools) more cleanly too.

Reply:
That does sound like a good convention. :+1: